tracing/src/main/java/com/palantir/tracing/DetachedSpan.java

```diff
@@ -138,6 +138,17 @@ default DetachedSpan childDetachedSpan(@Safe String operation) {
         return childDetachedSpan(operation, SpanType.LOCAL);
     }
 
+    /**
+     * Attaches the current {@link DetachedSpan} state to the current thread without creating additional spans.
+     * This is useful when a long-lived {@link DetachedSpan} measures many smaller operations (like async-io)
+     * in which we don't want to produce spans for each task, but do need tracing state associated for logging
+     * and potential child traces.
+     * @apiNote This must be executed within a try-with-resources block, and the parent detached span must still be
+     * completed separately.
+     */
+    @MustBeClosed
+    CloseableSpan attach();
```

Comment:
The `CloseableSpan` type is a little bit odd here because it's not actually creating a span. I played with a new (though functionally identical) CloseableContext type to make the naming cleaner, but it also made the api larger, so I've pushed reusing the existing type.